vtysh/vtysh.c

```diff
@@ -4244,10 +4244,24 @@ DEFUN (vtysh_show_daemons,
        "Show list of running daemons\n")
 {
 	unsigned int i;
+	struct vtysh_client *client;
+
+	for (i = 0; i < array_size(vtysh_client); i++) {
+		if (vtysh_client[i].flag == VTYSH_OSPFD) {
```

Comment:
I think the logic here is not quite right?
just walk the list of clients at each array index, testing their fds - no need to do anything specific for any particular proto.

Reply:
My recollection is that the vtysh_client array has a index/pointer for each daemon.  The client->next pointer is NULL for everything but the OSPF daemon ( see vtysh_update_all_instances ).  For the ospf client(s) the next pointer is used to walk the list of ospf clients that are connected. 

This does need to be cleaned up as that there is no need to test for VTYSH_OSPFD all you need to do is  just have a while client loop and this would work and it would be future proof for any other daemons that might add multiple versions of the same daemon.